Makefile

```
@@ -1,27 +1,69 @@
-# Makefile for Sphinx documentation
+GNUMAKEFLAGS := --no-builtins --no-builtin-variables
+.SUFFIXES:
 
-# You can set these variables from the command line.
-SPHINXOPTS    =
-SPHINXBUILD   = sphinx-build
-BUILDDIR      = site
+ifeq ($(OS),Windows_NT)
+--which := where
+--null := nul
+RM := rmdir /s /q
+else
+--which := which
+--null ?= /dev/null
+RM := rm -rf
+endif
+
+POETRY ?= $(shell $(--which) poetry 2> $(--null))
 
-# User-friendly check for sphinx-build
-ifeq ($(shell which $(SPHINXBUILD) >/dev/null 2>&1; echo $$?), 1)
-$(error The '$(SPHINXBUILD)' command was not found. Make sure you have Sphinx installed, then set the SPHINXBUILD environment variable to point to the full path of the '$(SPHINXBUILD)' executable. Alternatively you can add the directory with the executable to your PATH. If you don't have Sphinx installed, grab it from http://sphinx-doc.org/)
+ifeq ($(POETRY),)
+$(error Could not find the `poetry` command. \
+	Please make sure you have installed poetry, \
+	and that it is on your system's PATH environment variable. \
+	If you don't have poetry installed, \
+	please visit https://python-poetry.org \
+	for instructions on its installation)
 endif
 
-.PHONY: clean all
+SRCDIR ?= $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
+OUTDIR ?= $(join $(SRCDIR),_site)
+
+venv := $(notdir $(realpath $(shell $(POETRY) env info --path 2> $(--null))))
+
+archive.flags += --create --gzip --verbose
+archive.flags += --file=$(if $(ARCHIVE),$(ARCHIVE),cps-docs).tar.gz
```

If the user specifies `ARCHIVE`, wouldn't that be expected to include the extension?

Otherwise LGTM, thanks!

---

possibly. I could do the extra work of suffix detection/adding via `make`'s builtins for it if you think that's worth the time. Downside is technically someone could do `make archive ARCHIVE=docs.tar.bz2` and it wouldn't be a `bz2` 😅

---

IM(NS)HO there's no need to get that fancy. We generate a gzip'd tar; if you supply an `ARCHIVE` with an inappropriate extensions, that's on you. :slightly_smiling_face: I think it's less surprising though for `ARCHIVE` to be the full name and not just the name-without-extension. (Also, `ARCHIVE=foo.tgz` is fine.)

---

To make my position clearer :wink::
```suggestion
archive.flags += --file=$(if $(ARCHIVE),$(ARCHIVE),cps-docs.tar.gz)
```

---

Done and done then 🙂